pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
     return surf
 
 
+def SolidSphere(
+    outer_radius=0.5,
+    inner_radius=0.0,
+    radius_resolution=5,
+    start_theta=0.0,
+    end_theta=360.0,
+    theta_resolution=30,
+    start_phi=0.0,
+    end_phi=180.0,
+    phi_resolution=30,
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+):
+    """Create a solid sphere.
+
+    Parameters
+    ----------
+    outer_radius : float, default: 0.5
+        Outer radius of sphere.
+
+    inner_radius : float, default: 0.0
+        Inner radius of sphere.
+
+    radius_resolution : int, default: 5
+        Number of points in radial direction.
+
+    start_theta : float, default: 0.0
+        Starting longitude angle in degrees.
+
+    end_theta : float, default: 360.0
+        Ending longitude angle in degrees.
+
+    theta_resolution : int, default: 30
+        Number of points in theta direction.
+
+    start_phi : float, default: 0.0
+        Starting latitude angle in degrees.
+
+    end_phi : float, default: 180.0
+        Ending latitude angle in degrees.
+
+    phi_resolution : int, default: 30
+        Number of points in phi direction,
+        inclusive of polar axis if applicable.
+
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
```

Comment:
I'm a bit wary about this hiding user mistakes (i.e. a user explicitly passing `outer_radius` or `radius_resolution` _and_ `radius`, by mistake). The alternative I can think of is to set all the overrideable values to have `None` default, and raise if we have non-`None` values _and_ non-`None` `radius. But I know this would make the implementation a bit more complex, and the actual default values would be more detached from where they are documented.

Reply:
I am also worried about this.  I'm also worried about this problem `SolidSphere(radius=0.5)` <- raises an error.

Probably a lesser concern: the issue with burying the defaults only in the description is that the IDE then can't display the default and you have to hunt it down in the docstring that often doesn't render as well.  A small concern

A solution I thought of is to have separate functions (in this case functions that look like classes) for each which solves this problem, but creates a naming problem.

Reply:
Maybe there could be `NonUniformSolidSphere` which implements the general algorithm (and requires array inputs for all of radius/theta/phi) then `SolidSphere` is the "simple one" where users specify inner and outer radii, start/stop phi/theta etc.

Reply:
I like this idea.  My desire for ultra precision makes me want `UniformSolidSphere` and `SolidSphere`, where the former specifies inner and outer radii, etc.  But this would deviate from `Sphere`. I can implement and then we can argue over meaning later.

Reply:
Weak preference to leave the uniform version as `SolidSphere` since it will be straightforward for new users to gravitate to and test stuff out with (just like `Sphere` is now). The non-uniform version is only needed for niche use cases where the user knows it is more efficient to sample their data unevenly for some reason (like in my original use case where I know the features are tinier closer to the origin). But, even though the non-uniform version is cool and useful, I don't see it getting much use. But like I said, weak preference.

Reply:
I don't personally mind a generic helper with a more convoluted set of potential inputs, but multiple simpler helpers can work too. As long as they're all well linked from one another.

Reply:
I left `SolidSphere` to be the linearly sampled, according to spherical coordinates, version.  I named the flexible one `SolidSphereGeneric`.  I got tripped up trying to write a sentence like `The default parameters of SolidSphereNonuniform results in linearly sampled spherical coordinates.`  I think `Generic` gives a flavor of the flexibility of this approach.